src/internal/connector/data_collections.go

```diff
@@ -46,6 +46,10 @@ func (gc *GraphConnector) ProduceBackupCollections(
 		diagnostics.Index("service", sels.Service.String()))
 	defer end()
 
+	// Limit the max number of active requests to graph
+	ctrlOpts.ItemFetchParallelism = graph.Parallelism(sels.PathService()).
```

Comment:
I don't really see a case where we'd want to revert the value.  Do you?

Reply:
it's no so much about reverting, but making sure code for one part of the system doesn't pollute the values that code for another part of the system may see. To take an example, say we have backup for all SharePoint data types enabled. If we continue to have only a single control option then the different data types will fight for which value this is set to and also what value was passed into subsequent calls (i.e. accidentally use the output of OneDrive parameters as input for SP list parameters check because things are executed serially). If we scope it to downstream only (i.e. by returning a copy of the struct with updated values like with adding kv-pairs to context) then setting it for different services/data types can have different results which is what we'd want

Of course, we could add separate variables for each service/data type in the control options, but it seems more likely that could grow without bound and cause other problems

Reply:
Yeah, point.  Might make a tech debt for this.